embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/EssentialServiceModule.kt

```diff
@@ -277,7 +277,7 @@ internal class EssentialServiceModuleImpl(
     override val pendingApiCallsSender: PendingApiCallsSender by singleton {
         EmbracePendingApiCallsSender(
             networkConnectivityService,
-            apiRetryExecutor,
+            workerThreadModule.scheduledExecutor(ExecutorName.BACKGROUND_REGISTRATION),
```

Comment:
There's a KTLO task to look at our background thread usage to see if they're being used appropriately so we can consolidate threads to minimize unnecessary concurrent execution and maximize utilization. It's great to see this done incrementally - the retry checking job shouldn't be the same thread that requests are actually run on